pkg/api/v1alpha1/addons_types.go

```diff
@@ -247,15 +247,60 @@ type AddonsLayer struct {
 	Status AddonsLayerStatus `json:"status,omitempty"`
 }
 
-// +kubebuilder:object:root=true
-
 // AddonsLayerList contains a list of AddonsLayer.
+// +genclient
+// +kubebuilder:object:root=true
 type AddonsLayerList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []AddonsLayer `json:"items"`
 }
 
+// LayerMgrSpec defines the desired state of LayerMgr.
+type LayerMgrSpec struct {
```

Comment:
Whats the relationship between AddonLayer and LayerMgr?

Reply:
We decided to add a LayerMgr, still a WIP, to enable us to sync the phases of processing, i.e. prune and apply. So in the new implementation when an AddonsLayer is being reconciled and detects that a prune is required it can update the LayerMgr to set the phase to prune and that will cause the LayerMgr to be reconciled. In the LayerMgr reconcile processing it will update all AddonsLayers to force reprocessing and thus ensure everybody prunes. When all the AddonsLayers get to the state of pruned or beyond the LayerMgr will update the phase to deploy and requeue the AddonsLayers. In the AddonsLayer reconcile we will check the LayerMgr phase, if it is still prune then the reconcile of the AddonsLayer will not proceed beyond pruned. Still working on implementing this logic

Reply:
So the addonslayers will have an owner reference of LayerMgr?

Reply:
And will the LayerMgr have a list of addons layers?

Reply:
Not using owner ref, the AddonsLayer custom resources now have a group-name field that is the name of the LayerMgr which manages this group of AddonsLayers. When an AddonsLayer reconciler detects that this AddonsLayer requires pruning or applying it will use the k8sClient to get the LayerMgr with the name matching it's group-name and set the LayerMgr's phase to prune. When an AddonsLayer reconciler detects that the AddonLayer has pruned successfully (or no prune is required) it moves on to the deploy phase. Before it can do so it must check if the LayerMgr is still in prune phase. The LayerMgr reconciler will check all the AddonsLayers in its group and only set the phase to Deploying when all are pruned.

Reply:
Personally i'm not 100% convinced you need LayerMgr but you know the problem domain in a lot more detail. However, if you are going to have a LayerMgr and AddonsLayers it would be better to use one of the conventions/types/patterns already defined and make the relationship explicit. OwnerReference, corev1.ObjectReference, corev1.LocalObjectReference

Reply:
Making the group-name field in AddonsLayer an ObjectRef makes sense, thanks
The reason we are adding LayerMgr is to have a single place to orchestrate the layers, we started off without it but were concerned about race conditions with the addonsLayers trying to sync actions with each other

Reply:
Rajan clarified the spec so we don't need LayerMgr now